File: LeetCode/cpp/containsDuplicate.cpp
#include <vector>
#include <unordered_map>

class Solution {
    public:
        bool containsDuplicate(std::vector<int>& nums) {
            std::unordered_map<int, int> mapOfAppereance;

            for (int i = 0; i < nums.size(); i++)
            {
                mapOfAppereance[nums[i]]++;

                if (mapOfAppereance[nums[i]] > 1)
                {
                    return 1;
                }
            }
            
            return false;
        }   
};

File: LeetCode/py/containsDuplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        mapOfAppereance = {}

        for i in nums:
            if i in mapOfAppereance:
                mapOfAppereance[i] += 1
            else:
                mapOfAppereance[i] = 1

            if mapOfAppereance[i] > 1:
                return True

        return False